Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs

```diff
@@ -123,86 +152,99 @@ private void HandlePlacement()
         {
             if (IsBeingPlaced)
             {
-                SetLayerRecursively(transform, useDefaultLayer: false);
-                InputManager.Instance.AddGlobalListener(gameObject);
+                StartPlacing();
+            } else
+            {
+                StopPlacing();
+            }
+        }
 
-                // If the user is in placing mode, display the spatial mapping mesh.
-                if (AllowMeshVisualizationControl)
-                {
-                    SpatialMappingManager.Instance.DrawVisualMeshes = true;
-                }
+        private void StartPlacing()
+        {
+            var layerCacheTarget = PlaceParentOnTap ? ParentGameObjectToPlace : gameObject;
+            layerCacheTarget.SetLayerRecursively(IgnoreRaycastLayer, out layerCache);
+            InputManager.Instance.PushModalInputHandler(gameObject);
+
+            ToggleSpatialMesh();
 #if UNITY_WSA && !UNITY_EDITOR
 
-                //Removes existing world anchor if any exist.
-                WorldAnchorManager.Instance.RemoveAnchor(gameObject);
+            //Removes existing world anchor if any exist.
+            WorldAnchorManager.Instance.RemoveAnchor(gameObject);
```

Comment:
We should check the instance here too.  `WorldAnchorManager` should be optional.